llvm/lib/Analysis/AliasSetTracker.cpp

```diff
@@ -40,36 +40,37 @@ static cl::opt<unsigned>
                         cl::desc("The maximum number of pointers may-alias "
                                  "sets may contain before degradation"));
 
+bool AliasSet::isMustAliasMergeWith(AliasSet &AS, BatchAAResults &BatchAA) const {
+  for (const MemoryLocation &MemLoc : MemoryLocs)
+    for (const MemoryLocation &ASMemLoc : AS.MemoryLocs)
+      if (!BatchAA.isMustAlias(MemLoc, ASMemLoc))
```

Comment:
Must-alias is not a transitive relationship? Before we picked a representative and only checked those, why do we need to check all entries now?

Reply:
This is discussed in the RFC, using a representative also implies merging of the LocationSize, AAInfo,... of the other accesses into it: https://discourse.llvm.org/t/rfc-dont-merge-memory-locations-in-aliassettracker/73336#mustalias-sets-saturation-4

Reply:
Even though a representative is not possible, @nikic convinced me that any must-alias pair is sufficient because must-alias is transitive. See commit 5f82129f98f2.